ksql-engine/src/main/java/io/confluent/ksql/planner/plan/AggregateNode.java

```diff
@@ -143,7 +142,7 @@ public WindowExpression getWindowExpression() {
   private List<Pair<String, Expression>> getFinalSelectExpressions() {
     List<Pair<String, Expression>> finalSelectExpressionList = new ArrayList<>();
     if (finalSelectExpressions.size() != schema.fields().size()) {
-      throw new KsqlException(
+      throw new RuntimeException(
```

Comment:
Why change the type of exception here?

Reply:
IMO at some point we should have an exception hierarchy that at minimum categorizes our exceptions as being problems with inputs, external system errors, or internal errors (e.g. broken assumptions). I think for now we should try and stick to using KsqlException to indicate an input error (e.g. something was wrong with a statement) since that's overwhelmingly the most common reason KsqlException is raised. (Currently the only place where we try and make that distinction is the rest layer, which returns 400 when it sees a KsqlException). This condition is more like an internal error, so I changed it to raise something other than KsqlException.

Reply:
That's correct